prefect_gcp/cloud_storage.py

```diff
@@ -452,14 +452,216 @@ def example_cloud_storage_copy_blob_flow():
         dest_blob = source_blob
 
     source_blob_obj = source_bucket_obj.blob(source_blob)
-    partial_copy_blob = partial(
+    await run_sync_in_worker_thread(
         source_bucket_obj.copy_blob,
         blob=source_blob_obj,
         destination_bucket=dest_bucket_obj,
         new_name=dest_blob,
         timeout=timeout,
         **copy_kwargs,
     )
-    await to_thread.run_sync(partial_copy_blob)
 
     return dest_blob
+
+
+class GcsBucket(WritableDeploymentStorage, WritableFileSystem):
+    """
+    Block used to store data using GCP Cloud Storage Buckets.
+
+    Attributes:
+        bucket: Name of the bucket.
+        gcp_credentials: The credentials to authenticate with GCP.
+        bucket_folder: A default path to a folder within the GCS bucket to use
+            for reading and writing objects.
+
+    Example:
+        Load stored GCP Cloud Storage Bucket:
+        ```python
+        from prefect_gcp import GcsBucket
+        gcp_cloud_storage_bucket_block = GcsBucket.load("BLOCK_NAME")
+        ```
+    """
+
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/4CD4wwbiIKPkZDt4U3TEuW/c112fe85653da054b6d5334ef662bec4/gcp.png?h=250"  # noqa
+    _block_type_name = "GCS Bucket"
+
+    bucket: str = Field(..., description="Name of the bucket.")
+    gcp_credentials: GcpCredentials = Field(
+        ..., description="The credentials to authenticate with GCP."
+    )
```

Comment:
I think a `default_factory` is useful here for situations where you want to use the credentials from your environment.
```suggestion
    gcp_credentials: GcpCredentials = Field(
        default_factory=GcpCredentials, description="The credentials to authenticate with GCP."
    )
```